### PR TITLE
Add swap feature for date-range & time-range

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- https://github.com/the-deep/client/issues/2339#event-5931222444

## Changes:
- Add a swap feature for the date range inputs and time range inputs 

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/components/DateRangeInputWrapper/index.tsx [hunk 1]
```diff
@@ -33,6 +36,7 @@ function DateRangeInputWrapper<N extends string>(props: Props<N>) {
     } = props;
 
     const error = getErrorObject(riskyError);
+    const dateValue = !!value?.startDate && !!value?.endDate;
```
Comment (tnagorra) [inline app/components/DateRangeInputWrapper/index.tsx]:
```suggestion
    const hasDateValue = !!value?.startDate || !!value?.endDate;
```

File: app/components/TimeRangeInputWrapper/index.tsx [hunk 2]
```diff
@@ -33,6 +36,7 @@ function TimeRangeInputWrapper<N extends string>(props: Props<N>) {
     } = props;
 
     const error = getErrorObject(riskyError);
+    const timeValue = !!value?.startTime && !!value?.endTime;
```
Comment (tnagorra) [inline app/components/TimeRangeInputWrapper/index.tsx]:
```suggestion
    const hasTimeValue = !!value?.startTime || !!value?.endTime;
```

File: app/components/TimeRangeInputWrapper/index.tsx [hunk 3]
```diff
@@ -66,6 +74,18 @@ function TimeRangeInputWrapper<N extends string>(props: Props<N>) {
                 disabled={disabled}
                 error={error?.endTime}
             />
+            {timeValue && (
+                <Button
+                    name={undefined}
+                    variant="action"
+                    onClick={handleSwapTimeRange}
+                    disabled={disabled}
+                    readOnly={readOnly}
```
Comment (tnagorra) [inline app/components/TimeRangeInputWrapper/index.tsx]: Let's hide this button if readOnly is true

File: app/components/DateRangeInputWrapper/index.tsx [hunk 4]
```diff
@@ -66,6 +74,18 @@ function DateRangeInputWrapper<N extends string>(props: Props<N>) {
                 disabled={disabled}
                 error={error?.endDate}
             />
+            {dateValue && (
+                <Button
+                    name={undefined}
+                    variant="action"
+                    onClick={handleSwapDateRange}
+                    disabled={disabled}
+                    readOnly={readOnly}
```
Comment (tnagorra) [inline app/components/DateRangeInputWrapper/index.tsx]: Let's hide this button if readOnly is true

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/components/DateRangeInputWrapper/index.tsx [hunk 5]
```diff
@@ -74,13 +74,12 @@ function DateRangeInputWrapper<N extends string>(props: Props<N>) {
                 disabled={disabled}
                 error={error?.endDate}
             />
-            {dateValue && (
+            {hasDateValue && (
```
Comment (tnagorra) [inline app/components/DateRangeInputWrapper/index.tsx]: Let's hide this button if readOnly is true

File: app/components/TimeRangeInputWrapper/index.tsx [hunk 6]
```diff
@@ -74,13 +74,12 @@ function TimeRangeInputWrapper<N extends string>(props: Props<N>) {
                 disabled={disabled}
                 error={error?.endTime}
             />
-            {timeValue && (
+            {hasTimeValue && (
```
Comment (tnagorra) [inline app/components/TimeRangeInputWrapper/index.tsx]: Let's hide this button if readOnly is true

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**